.github/workflows/main.yml

```
@@ -0,0 +1,196 @@
+name: Main
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
+  NODE_VERSION: 18
+  SOLANA_VERSION: 1.18.12
+  CARGO_CACHE: |
+    ~/.cargo/bin/
+    ~/.cargo/registry/index/
+    ~/.cargo/registry/cache/
+    ~/.cargo/git/db/
+    target/
+
+jobs:
+  build_programs:
+    name: Build programs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v4
+      - name: Setup environment
+        uses: ./.github/actions/setup
+        with:
+          node: ${{ env.NODE_VERSION }}
+          solana: ${{ env.SOLANA_VERSION }}
+      - name: Cache cargo dependencies
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.CARGO_CACHE }}
+          key: ${{ runner.os }}-cargo-programs-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-programs
+      - name: Build programs
+        run: pnpm programs:build
```

I think we want the `bpf-entrypoint` feature here, right?

```suggestion
        run: pnpm programs:build --features bpf-entrypoint
```

---

Yes, I was planning in fixing the CI in a separate PR, once it is setup on the repo. CI won't run on the branch until there is a workflow on main.